docs/tutorial/plugins.md

```diff
@@ -0,0 +1,547 @@
+---
+title: Create Plugins in JavaScript
+description: Plugins implemented in JavaScript are easily used across different projects, as they do not require any additional programs to be installed.
+---
+
+Jupyter Book uses the [MyST plugins infrastructure](xref:guide/plugins) to extend Jupyter Book's base functionality.
+Here are some things you can do with plugins:
+
+- Define a new directive or role to generate block-level or inline content.
+- Define a transform that modifies a page's content at build time.
+
+Plugins are natively written in JavaScript, or by writing an executable script in another language that follows [the MyST plugin protocol](xref:guide/executable-plugins).
+This tutorial shows you how to create a JavaScript plugin from scratch.
+
+## Create a basic Jupyter Book template
+
+We'll start by creating a basic template book that we can add to.
+
+🛠 First, create a new folder for a demo Jupyter Book (or use a pre-existing book).
+
+```shell
+$ mkdir plugin-demo
+$ cd plugin-demo
+```
+
+🛠 Next initialize a Jupyter Book in the folder to add a configuration file:
+
+```shell
+$ jupyter-book init
+```
+
+🛠 Finally, create an empty file that we'll use to play around with our plugin:
+
+```shell
+$ echo '# My demo page' > myplugindemo.md
+```
+
+🛠 And add it to your Jupyter Book's configuration file:
+
+```yaml
+project:
+  toc:
+    - file: myplugindemo.md
+```
+
+You should now be able to run `myst start` and see a page with just your title rendered:
+
+🛠️ Run `myst start` to confirm that your demo book is set up properly.
+
+```shell
+$ myst start
+```
+
+## Create an empty plugin script
+
+Next we'll create an empty JavaScript file[^esm] that we'll use to add new plugin logic.
+
+[^esm]: The format of the Javascript should be an ECMAScript modules, not CommonJS. This means it uses `import` statements rather than `require()` and is the most modern style of Javascript. See [the MDN modules documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules) for background on modules.
```

Comment:
```suggestion
[^esm]: The format of the JavaScript should be an ECMAScript modules, not CommonJS. This means it uses `import` statements rather than `require()` and is the most modern style of JavaScript. See [the MDN modules documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules) for background on modules.
```